### PR TITLE
Add Profile page with user login/register

### LINKED ISSUE #57: Add Profile page
## Description
As a user that is not logged in, I want to be able to register or sign in. When I'm logged in, I'd like to see my profile and an option to logout

## Design
[Link](https://www.figma.com/file/L4dBE9w4Hc6a6e6PLQiNC9/Perfect-Pic?node-id=a2%3A3)

## Acceptance criteria

- [x] check login hook for all pages by jwt cookie
- [x] show the register/login screen on the profile page if not logged in
- [x] login or create a user with submit
- [x] show the profile and logout CTA if logged in 

## Blocked by
#54

## Size
M

### PR DESCRIPTION
**Pull request by dirkzukunft**
Resolves #57

https://perfectpic-dirkzukunft--vgd1r2.herokuapp.com/profile

### CODE REVIEW

File: src/app/hooks/useExistsUser.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import axios from 'axios';
+import { useQuery } from 'react-query';
+
+export default function useExistsUser(username: string): boolean | null {
```
Comment (marwinburesch) [inline src/app/hooks/useExistsUser.ts:4]: `useUserExists` sponsored by @lmachens

File: package.json [hunk 2]
```diff
@@ -53,17 +54,22 @@
...
     "jsonwebtoken": "^8.5.1",
+    "jwt-decode": "^3.1.2",
     "mongodb": "^4.1.2",
     "node-fetch": "2.6.2",
     "react": "^17.0.2",
     "react-device-detect": "^1.17.0",
     "react-dom": "^17.0.2",
     "react-masonry-component": "^6.3.0",
+    "react-password-strength-bar": "^0.3.5",
+    "react-query": "^3.24.4",
```
Comment (marwinburesch) [inline package.json:72]: omg Tanner Lindsey

File: src/app/pages/Profile/Profile.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      } catch (error) {
+        setFormError('Wrong password');
+      }
+    }
+  }
+
+  function logout() {
+    cookies.remove('auth');
+
+    queryClient.setQueryData(['checkUser', currentUser], true);
```
Comment (marwinburesch) [inline src/app/pages/Profile/Profile.tsx:60]: I don't understand why you only interact with `queryClient` on logout

Comment (dirkzukunft) [reply src/app/pages/Profile/Profile.tsx:60]: It deletes the cached value of the useExistsUser hook. It's for the scenario: I register a new user, login, and logout within one session. When I return to the register/login screen, I need the refetch.

File: package.json [hunk 4]
```diff
@@ -53,17 +54,22 @@
...
+    "js-cookie": "^3.0.1",
     "jsonwebtoken": "^8.5.1",
+    "jwt-decode": "^3.1.2",
     "mongodb": "^4.1.2",
     "node-fetch": "2.6.2",
     "react": "^17.0.2",
     "react-device-detect": "^1.17.0",
     "react-dom": "^17.0.2",
     "react-masonry-component": "^6.3.0",
+    "react-password-strength-bar": "^0.3.5",
```
[the remark this answers is not in the document]
Comment (dirkzukunft) [reply package.json:71]: I know, the trade off was to skip it or take this ;) 
Issue to refactor later.

File: src/app/pages/Profile/Profile.tsx (hunk 3 above)
Comment (dirkzukunft) [reply src/app/pages/Profile/Profile.tsx]: Replaced by `queryClient.invalidateQueries();`

**Pull request merged**